kafka/client_async.py

```diff
@@ -635,6 +635,16 @@ def _poll(self, timeout):
             self._sensors.select_time.record((end_select - start_select) * 1000000000)
 
         for key, events in ready:
+            if key.fileobj.fileno() < 0:
```

Comment:
Would de-registering the offending socket not also work in this case, while forcing the client to try and refresh the connection?

Reply:
This is probably the least intrusive way to fix the issue on 3rd party package. I don't know kafka-python very well but I can dig deeper to see how kafka-python handles sockets, however we may still need the 0.1s sleep, otherwise the busy-retry may happen again because this part of the code runs in a separate thread, in an infinite loop. Thank you.

Reply:
Updated the commit to unregister the socket instead. Thank you.